x-pack/plugins/security_solution/public/timelines/components/fields_browser/field_items.tsx

```diff
@@ -56,11 +62,134 @@ Description.displayName = 'Description';
  * An item rendered in the table
  */
 export interface FieldItem {
+  ariaRowindex?: number;
+  checkbox: React.ReactNode;
   description: React.ReactNode;
   field: React.ReactNode;
   fieldId: string;
 }
 
+const DraggableFieldsBrowserFieldComponent = ({
+  browserFields,
+  categoryId,
+  fieldCategory,
+  fieldName,
+  highlight = '',
+  onUpdateColumns,
+  timelineId,
+  toggleColumn,
+}: {
+  browserFields: BrowserFields;
+  categoryId: string;
+  fieldCategory: string;
+  fieldName: string;
+  highlight?: string;
+  onUpdateColumns: OnUpdateColumns;
+  timelineId: string;
+  toggleColumn: (column: ColumnHeaderOptions) => void;
+}) => {
+  const keyboardHandlerRef = useRef<HTMLDivElement | null>(null);
+  const [closePopOverTrigger, setClosePopOverTrigger] = useState<boolean>(false);
+  const [hoverActionsOwnFocus, setHoverActionsOwnFocus] = useState<boolean>(false);
+
+  const handleClosePopOverTrigger = useCallback(() => {
+    setClosePopOverTrigger((prevClosePopOverTrigger) => !prevClosePopOverTrigger);
+
+    setHoverActionsOwnFocus((prevHoverActionsOwnFocus) => {
+      if (prevHoverActionsOwnFocus) {
+        // on the next tick, re-focus the keyboard handler if the hover actions owned focus
+        setTimeout(() => {
+          keyboardHandlerRef.current?.focus();
+        }, 0);
+      }
+      return false; // always give up ownership
+    });
+
+    setTimeout(() => {
+      setHoverActionsOwnFocus(false);
+    }, 0); // invoked on the next tick, because we want to restore focus first
+  }, [keyboardHandlerRef, setClosePopOverTrigger, setHoverActionsOwnFocus]);
```

Comment:
nit -> no needs of these dependencies setClosePopOverTrigger, setHoverActionsOwnFocus